README.md

```diff
@@ -55,6 +55,57 @@ If you fail to launch a pre-built binary, please try the second way.
 * ctrl + f, F3, shift+F3: find a string 
 * F1, ctrl+shift+p: open a command palette
 
+### Annotate pipelines with function names
+
+
+1. Prepare the input file:
+
+```sh
+# Extract used addresses:
+cat m5out/trace.out | grep -oe '^O3PipeView:fetch:[0-9]\+:0x[0-9a-f]\+:' | grep -oe '0x[0-9a-f]\+' | uniq | sort | uniq > instructions.log
+
+
+# Extract address info
+while read addr ; do
+    info=$(addr2line -e <BINARY WITH DEBUG INFO> -f -C -i -s -p "$addr")
+    echo -e "$addr\n$info\n\n"
+done < instructions.log
+
```

Comment:
this could be done in a more efficient way... might update later as it will affect awk below